src/runtime/client-event-manager-test.js

```diff
@@ -34,272 +35,310 @@ const DEFAULT_EVENT = {
-        throw errorSent;
+        expect(logStub).to.have.been.calledWith(DEFAULT_ERROR);
+      });
+
+      it('should log filterer errors to the console', async () => {
+        eventManager.registerEventFilterer(() => {
+          throw DEFAULT_ERROR;
+        });
+
+        eventManager.logEvent(DEFAULT_EVENT);
+        await tick();
+
+        expect(logStub).to.have.been.calledWith(DEFAULT_ERROR);
       });
-      eventMan.logEvent(DEFAULT_EVENT);
-      await eventMan.lastAction_;
-      expect(errorReceived).to.equal(errorSent);
     });
   });
 
   describe('event management', () => {
-    let eventMan;
-    let counter;
+    let eventManager;
+    let events;
 
     beforeEach(() => {
-      eventMan = new ClientEventManager(RESOLVED_PROMISE);
-      counter = 0;
-      eventMan.registerEventListener(() => counter++);
+      eventManager = new ClientEventManager(RESOLVED_PROMISE);
+      events = [];
+      eventManager.registerEventListener(event => events.push(event));
+    });
+
+    it('throws if listener is not a function', async () => {
+      expect(() => {
+        eventManager.registerEventListener('Hey guys');
+      }).to.throw(/must be a function/);
     });
 
-    it('should be able to listen for events', async () => {
-      //verify it supports 1 listener
+    it('supports a single listener', async () => {
+      eventManager.logEvent(DEFAULT_EVENT);
+      await tick();
+      expect(events).to.deep.equal([DEFAULT_EVENT]);
+    });
+
+    it('supports additional listeners', async () => {
+      eventManager.registerEventListener(event => events.push(event));
 
-      eventMan.logEvent(DEFAULT_EVENT);
-      await eventMan.lastAction_;
-      expect(counter).to.equal(1);
+      eventManager.logEvent(DEFAULT_EVENT);
+      await tick();
+      expect(events).to.deep.equal([DEFAULT_EVENT, DEFAULT_EVENT]);
+    });
 
-      //verify it supports multiple listeners
-      eventMan.registerEventListener(() => counter++);
-      eventMan.logEvent(DEFAULT_EVENT);
-      await eventMan.lastAction_;
-      expect(counter).to.equal(3);
+    it('should throw if filterer is not a function', async () => {
+      expect(() => {
+        eventManager.registerEventFilterer(null);
+      }).to.throw(/must be a function/);
     });
 
     it('should be able to filter out some events', async () => {
-      //filter out the default origin
-      eventMan.registerEventFilterer(event =>
+      // Filter out the default origin.
+      eventManager.registerEventFilterer(event =>
         event.eventOriginator === DEFAULT_ORIGIN
           ? EventManagerApi.FilterResult.CANCEL_EVENT
           : EventManagerApi.FilterResult.PROCESS_EVENT
       );
 
-      //ensure the default origin is filtered out
-      eventMan.logEvent(DEFAULT_EVENT);
-      await eventMan.lastAction_;
-      expect(counter).to.equal(0);
-
-      //ensure the other origin is not filtered out
-      DEFAULT_EVENT.eventOriginator = OTHER_ORIGIN;
```

Comment:
Looks like this was mutating a `const` shared across most the tests

Reply:
It was mutating then unmutating a sub value of the const which is valid but potentially slightly confusing.

Reply:
e.g. 
const x = {a: 1};
x.a = 5;

is valid but potentially confusing

Reply:
I don't see the `DEFAULT_EVENT` object being mutated again in this function

Reply:
I think we should avoid mutating shared state across tests (if possible), since that can cause flakiness